xresolver/consul/consullistener.go

```diff
@@ -0,0 +1,103 @@
+package consul
+
+import (
+	"context"
+	"errors"
+	"github.com/Comcast/webpa-common/logging"
+	"github.com/Comcast/webpa-common/service/monitor"
+	"github.com/Comcast/webpa-common/xresolver"
+	"github.com/go-kit/kit/log"
+	"regexp"
+)
+
+var find = regexp.MustCompile("(.*)" + regexp.QuoteMeta("[") + "(.*)" + regexp.QuoteMeta("]") + regexp.QuoteMeta("{") + "(.*)" + regexp.QuoteMeta("}"))
+
+type Options struct {
+	// Watch is what to url to match with the consul service
+	// exp. { "beta.google.com" : "caduceus" }
+	Watch map[string]string `json:"watch"`
+
+	Logger log.Logger `json:"-"`
+}
+
+type ConsulWatcher struct {
+	debugLogger log.Logger
+	infoLogger  log.Logger
+	warnLogger  log.Logger
+	errorLogger log.Logger
+
+	config *Options
+
+	watch     map[string]string
+	balancers map[string]*xresolver.RoundRobin
+}
+
+func NewConsulWatcher(o *Options) *ConsulWatcher {
```

Comment:
Don't pass `Options` as a pointer here.  This is a potential race condition, given what happens immediately after this line.  The `Logger` field is edited on a shared object.

In general, just pass `Options`-style structs by value.  I know other places have them passed by pointer, but this was due to a lack of familiarity with concurrent programming.